cmd/sink/filters/filters.go

```diff
@@ -128,38 +127,87 @@ func (f *Filters) createFilters(
 	if err != nil {
 		return archiveMap, deleteMap, archiveOnDeleteMap, err
 	}
+
 	var errList []error
 	for _, resource := range resources {
 		gvk := schema.FromAPIVersionAndKind(resource.Selector.APIVersion, resource.Selector.Kind)
 		namespaceGvk := NamespaceGVKFromNamespaceAndGvk(namespace, gvk)
-		if resource.ArchiveWhen != "" {
-			archiveExpr, err := ocel.CreateCelExprOr(resource.ArchiveWhen, globalArchive[gvk.String()])
-			if err != nil {
-				errList = append(errList, err)
-			} else {
-				archiveMap[namespaceGvk] = *archiveExpr
-			}
+
+		errs := addLocalFilters(resource.ArchiveWhen, globalArchive[gvk], archiveMap, namespaceGvk)
+		if len(errs) > 0 {
```

Comment:
Do we even need this check?  Why not just always append?

Reply:
Nice, I like it. Done.